pkg/autodiscovery/common/utils/kubelet.go

```diff
@@ -16,3 +20,35 @@ func GetCustomCheckID(annotations map[string]string, containerName string) (stri
 	id, found := annotations[fmt.Sprintf(checkIDAnnotationFormat, containerName)]
 	return id, found
 }
+
+// ValidateAnnotationsMatching detects if AD annotations don't match a valid container identifier
+func ValidateAnnotationsMatching(annotations map[string]string, containerIdentifiers map[string]bool, containerNames map[string]bool, adPrefix string) []error {
+	var errors []error
+	adAnnotation := fmt.Sprintf(`%s.+\..+`, adPrefix)
+	checkIDAnnotation := fmt.Sprintf(checkIDAnnotationFormat, ".+\\")
+
+	for annotation := range annotations {
+		if matched, _ := regexp.MatchString(checkIDAnnotation, annotation); matched {
+			// validate check.id annotation
+			err := validateIdentifier(annotation, containerNames, adPrefix)
```

Comment:
If `adPrefix != "ad.datadoghq.com/"` (or at least has different length), then `validateIdentifier` will fail to extract container identifier from the check id annotation like `ad.datadoghq.com/nginx.check.id`

Reply:
Will `ad.datadoghq.com/nginx.check.id` be parsed correctly if `adPrefix` is set to something else (as in `Legacy annotations` test case)?

Reply:
If only the legacy annotations are used with `check.id`, it wouldn't match the first if statement checking for `check.id`, but it would show a misleading error message about using an invalid identifier on the second if statement since it contains `.`. I spoke to someone on Container Integrations and they suggested only checking the new pod annotations format as it's rare to see someone using the legacy annotations. I'll also add to the tests to check that the legacy annotations for `check.id` are ignored.